### PR TITLE
Make snarkyjs an ES module

### PR DESCRIPTION
**Pull request by mitschabaude**
* companion of https://github.com/o1-labs/snarkyjs/pull/366

### CODE REVIEW

File: scripts/update-snarkyjs-bindings.sh [hunk 1]
```diff
@@ -5,25 +5,27 @@ SNARKY_JS_PATH=src/lib/snarky_js_bindings/snarkyjs
...
-sed -i 's/function invalid_arg(s){throw \[0,Invalid_argument,s\]/function invalid_arg(s){throw joo_global_object.Error(s.c)/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.js
-sed -i 's/return \[0,Exn,t\]/return joo_global_object.Error(t.c)/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.js
-sed -i 's/function raise(t){throw caml_call1(to_exn$0,t)}/function raise(t){throw Error(t?.[1]?.c ?? "some error")}/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.js
+sed -i 's/function failwith(s){throw \[0,Failure,s\]/function failwith(s){throw joo_global_object.Error(s.c)/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.cjs
+sed -i 's/function invalid_arg(s){throw \[0,Invalid_argument,s\]/function invalid_arg(s){throw joo_global_object.Error(s.c)/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.cjs
+sed -i 's/return \[0,Exn,t\]/return joo_global_object.Error(t.c)/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.cjs
+sed -i 's/function raise(t){throw caml_call1(to_exn$0,t)}/function raise(t){throw Error(t?.[1]?.c ?? "some error")}/' "$SNARKY_JS_PATH"/src/node_bindings/snarky_js_node.bc.cjs
```
Comment (MartinMinkov) [inline scripts/update-snarkyjs-bindings.sh:20]: I've found this error to be really unhelpful and hard to track down. Maybe you can indicate that the error is being caught in this function and thrown again?

e.g.
`Unknown error caught while building SnarkyJS and raised in update-snarkyjs-bindings.sh.`.

File: scripts/update-snarkyjs-bindings.sh [hunk 2]
```diff
@@ -5,25 +5,27 @@ SNARKY_JS_PATH=src/lib/snarky_js_bindings/snarkyjs
...
 
 # optimize wasm / minify JS (we don't do this with jsoo to not break the error message fix above)
-pushd "$SNARKY_JS_PATH"/src/node_bindings
-  wasm-opt --detect-features --enable-mutable-globals -O4 plonk_wasm_bg.wasm -o plonk_wasm_bg.wasm.opt
-  mv plonk_wasm_bg.wasm.opt plonk_wasm_bg.wasm
-  npx esbuild --minify --log-level=error snarky_js_node.bc.js > snarky_js_node.bc.min.js
-  mv snarky_js_node.bc.min.js snarky_js_node.bc.js
-popd
+# pushd "$SNARKY_JS_PATH"/src/node_bindings
```
Comment (MartinMinkov) [inline scripts/update-snarkyjs-bindings.sh:23]: Why leave this in if we're just commenting it out?

**Review by MartinMinkov — APPROVED** [review]
LGTM other than a few points!

File: scripts/update-snarkyjs-bindings.sh (hunk 1 above)
Comment (mitschabaude) [reply scripts/update-snarkyjs-bindings.sh:20]: Good point, will change it! (The error is not caught while building though)

File: scripts/update-snarkyjs-bindings.sh (hunk 2 above)
Comment (mitschabaude) [reply scripts/update-snarkyjs-bindings.sh:23]: hm I guess you're right. I wanted to see whether not minifying the build makes the DX better -- there's trade-offs, so I'm not quite sure about it

**Review by bkase — APPROVED** [review]

**Pull request merged**